packages/dialog/theme/material/vaadin-dialog-styles.js

```diff
@@ -10,29 +10,98 @@ const dialogOverlay = css`
     max-width: 560px;
     min-width: 280px;
     -webkit-tap-highlight-color: transparent;
+    border-radius: 8px;
+  }
+
+  [part='overlay']:focus-visible {
+    box-shadow: 0 0 0 2px var(--material-primary-color), var(--material-shadow-elevation-24dp);
   }
 
   [part='content'] {
     padding: 24px;
   }
 
-  :host([has-header]) [part='header'],
-  :host([has-title]) [part='header'] {
-    padding: 9px 24px;
-    font-weight: 500;
+  :host(:is([has-header], [has-title])) [part='header'] {
+    padding: 16px;
+  }
+
+  :host(:is([has-header], [has-title])) [part='header'] + [part='content'] {
+    padding-top: 0;
+  }
+
+  [part='header'],
+  [part='header-content'],
+  [part='footer'] {
     gap: 8px;
-    border-bottom: 1px solid var(--material-divider-color);
+    line-height: 1.2;
+  }
+
+  [part='title'] {
+    font-size: var(--material-h5-font-size);
+    font-weight: 500;
+    margin-inline-start: 8px;
   }
 
   :host([has-footer]) [part='footer'] {
     padding: 8px;
-    border-top: 1px solid var(--material-divider-color);
+  }
+
+  @media (min-height: 320px) {
+    :host(:is([has-header], [has-title])) [part='header'] {
+      box-shadow: 0 1px 0 0 var(--material-divider-color);
+    }
+
+    :host([has-footer]) [part='footer'] {
+      box-shadow: 0 -1px 0 0 var(--material-divider-color);
+    }
+
+    /* "scroll divider" */
+
+    :host([has-footer]) [part='content'] {
+      height: 100%;
+      padding-bottom: 0;
+    }
+
+    :host([has-footer]) [part='content'] slot {
+      display: block;
+      min-height: calc(100% - 26px);
+    }
```

Comment:
This part can seem quite “unorthodox”, using the `<slot>` element for styling purposes. You could even say this is relying on private API, and should not be used.

What I would really like, instead, is to have a way to identify with CSS when the content is overflowing in either direction (above or below), so there would be a way to implement more complex scroll-based styles for the header and footer than just a 1px divider. For example, apply a background color or a box-shadow.

Reply:
What I’m trying to say, is maybe you want to convince me that this current implementation is something we would not want to maintain 😅

Reply:
I'm ok to keep this for now as it's unlikely that we'll change Material theme again any time soon.

> What I would really like, instead, is to have a way to identify with CSS when the content is overflowing in either direction (above or below), so there would be a way to implement more complex scroll-based styles

IMO we could add this logic to `vaadin-scroller` and use it internally for scrollable Dialog content.